### DISCUSSION TITLE
Boundary Condition not being obeyed

### DISCUSSION THREAD

**Discussion by gj19866** [idaholab/moose#26812] [Q&A General]
Hi there,

I have 2 variables, and I want to apply Neumann BCs to them. 
I want that $\nabla \varphi \cdot \hat{n} = 0$, and that $\nabla \psi \cdot \hat{n} = j$, where j is a constant.

When I try to implement this I get that $\nabla \psi \cdot \hat{n} = j$ is obeyed, however $\nabla \varphi \cdot \hat{n} = 0$ is not obeyed. 
When I use a DirichletBC $\varphi = 0$ the solution looks how I expect it to (0 everywhere). This also then has $\nabla \varphi \cdot \hat{n} = 0$.

Is there a reason you can think of why my $\nabla \varphi \cdot \hat{n} = 0$  boundary is not being obeyed properly? Below is an image of what I'm seeing.

![image](https://github.com/idaholab/moose/assets/141067346/8398fd09-1378-4c79-b588-7ac843b96392)

All the best,
Gillian

**Comment by schakrabortygithub** [comment]
If it is a transient problem, you might be missing initial condition. Can you try by specifying non-zero initial_condition e.g. 'Constant_IC', and zero or finite_value Neumann BC.
`

**Comment by GiudGiud** [comment]
Which boundary did you apply a zero current boundary condition? both?
Did you specify a Neumann BC or did you just specify no boundary conditions on that boundary?
